### PR TITLE
Prototype builder update

### LINKED ISSUE #7212: 4.x MediaContext requires .build()
* Helidon Version: 4.0.0-ALPHA6
----------

The following does NOT include the default media support that is registered by `io.helidon.nima.http.media.MediaContextBuilderInterceptor`.

```java
 WebServer.builder().mediaContext(MediaContext.builder());
```

Instead you get a `MediaContext` that is empty and unable to read or write any entity types.

However if you invoke `.build()` everything works ok.

```java
WebServer.builder().mediaContext(MediaContext.builder().build());
```

The issue is that the builder implements the config type, and the config type is consumed by other parts.

```
// this works because the builder implements MediaContextConfig
// not because the method is overloaded to accept Supplier<MediaContextConfig>
// and the call the .get() is equivalent to .build()
WebServer.builder().mediaContext( MediaContext.builder());
```

### LINKED ISSUE #7283: 4.x: Builder interceptor to decorator
Builder interceptor - either change the generated code to use the return value, or change the interface and name to only accept builder instance (to allow mutation, not replacement)

### PR DESCRIPTION
**Pull request by tomas-langer**
Fixes #7212 
Fixes #7283 

Also changed interceptor to decorator, as interceptor not needed for builders

### CODE REVIEW

**Comment by tomas-langer** [comment]
Rebased on main

**Review by romain-grecourt — COMMENTED** [review]
The following does not compile anymore.

```java
WebServer.builder().mediaContext(MediaContext.builder());
```

The builder should have a variant of the `mediaContext` method that accepts `Supplier<MediaContext>`.

File: nima/webclient/api/src/main/java/io/helidon/nima/webclient/api/HttpClientConfigBlueprint.java [hunk 1]
```diff
@@ -151,7 +151,7 @@ default ClientRequestHeaders defaultRequestHeaders() {
      *
      * @return media context
      */
-    @ConfiguredOption
+    @ConfiguredOption("create()")
```
Comment (romain-grecourt) [inline nima/webclient/api/src/main/java/io/helidon/nima/webclient/api/HttpClientConfigBlueprint.java:154]: Is "create()" a config key  ?

File: nima/webclient/http1/src/main/java/io/helidon/nima/webclient/http1/Http1ClientConfigBlueprint.java [hunk 2]
```diff
@@ -17,17 +17,19 @@
...
  */
-@Prototype.Blueprint(builderInterceptor = Http1ClientConfigSupport.Http1BuilderInterceptor.class)
+@Prototype.Blueprint
 interface Http1ClientConfigBlueprint extends HttpClientConfig, Prototype.Factory<Http1Client> {
     /**
      * HTTP/1.1 specific configuration.
      *
      * @return protocol specific configuration
      */
+    @ConfiguredOption("create()")
```
Comment (romain-grecourt) [inline nima/webclient/http1/src/main/java/io/helidon/nima/webclient/http1/Http1ClientConfigBlueprint.java:33]: Is "create()" a config key  ?

File: nima/webclient/api/src/main/java/io/helidon/nima/webclient/api/HttpClientConfigBlueprint.java (hunk 1 above)
Comment (tomas-langer) [reply nima/webclient/api/src/main/java/io/helidon/nima/webclient/api/HttpClientConfigBlueprint.java:154]: create() is a static method on the type - this is the default value

Comment (tomas-langer) [reply nima/webclient/api/src/main/java/io/helidon/nima/webclient/api/HttpClientConfigBlueprint.java:154]: create() is a static method on the type - this is the default value

**Comment by tomas-langer** [comment]
I would prefer to add the methods that accept supplier as a follow up to this issue. This is fixing the problem of sending builder where we should have sent a built value.

**Review by romain-grecourt — APPROVED** [review]

**Pull request merged**